Proiect_PAO/src/aplicatieBancara/Card/Card.java

```diff
@@ -0,0 +1,108 @@
+package aplicatieBancara.Card;
+
+import aplicatieBancara.Cont.Cont;
+import aplicatieBancara.TipCard;
+
+import java.nio.charset.StandardCharsets;
+import java.util.*;
+
+
+public abstract class Card {
+
+    protected final int cardId;
+    protected final int CVV;
+    protected String numar;
+    protected String IBAN;
+    protected double sold;
+    protected Cont cont;
+    protected final Date dataExpirare;
+    //protected String PIN;
+
+
+    static private final HashSet<String> usedNumbers = new HashSet<String>();
+
+    abstract public String getTip();
+
+    public Card(int cardId, Cont cont) {
+        this.cardId = cardId;
+        this.IBAN = cont.getIBAN();
+        this.cont = cont;
+        this.numar = this.generareNumarCard();
+
+        //generare numar card
+        while(usedNumbers.contains(this.numar))
+            this.numar = this.generareNumarCard();
+        usedNumbers.add(this.numar);
+
+        //generare CVV
+        this.CVV = this.generareCVV();
+
+        //generare data expirare
+        Calendar c = Calendar.getInstance();
```

Comment:
Calendar nu prea mai este folosit, chiar daca nu este gresit. java.time is the best package from java pe care il poti folosi ceva de genul: LocalDate expirationDate = LocalDate.now().plusYears(3);